### PR TITLE
refactor(generative_ai): (set-3) Reduced hidden variables and refactored code in GenAI Samples

### PR DESCRIPTION
**Pull request by Thoughtseize1**
## Description
**Summary of Changes**
Simplified sample code: Updated code comments and removed unnecessary variables. Declared global variables where appropriate. Changed tests. 

**Major changes:**

- `gemini_rapid_evaluation.py`: Removed the model parameter from the evaluation task because the dataset includes pre-existing responses. This change allows the evaluation to be performed in Bring Your Own Prediction (BYOP) mode, using the provided responses to calculate metrics directly.
- `gemini_rapid_evaluation_test.py`:  Attempting to run tests that were previously skipped.

Fixes #<ISSUE-NUMBER>

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [x] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [ ] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file)
- [x] **Tests** pass:   `nox -s py-3.9` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [x] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: generative_ai/code_generation_function.py [hunk 1]
```diff
@@ -18,9 +18,9 @@ def generate_a_function() -> object:
     # [START generativeaionvertexai_sdk_code_generation_function]
     from vertexai.language_models import CodeGenerationModel
 
-    # TODO(developer): update temperature value. Ex: temperature = 0.5
+    # TODO(developer): update temperature value. Ex: temperature = 0.1
```
Comment (msampathkumar) [inline generative_ai/code_generation_function.py:21]: Do we need this `TODO` ?

File: generative_ai/gemini_reasoning_engine.py [hunk 2]
```diff
@@ -11,39 +11,37 @@
...
 ) -> reasoning_engines.ReasoningEngine:
     # [START generativeaionvertexai_create_reasoning_engine_basic]
     import vertexai
     from vertexai.preview import reasoning_engines
 
-    # TODO(developer): Update and un-comment below lines
-    # project_id = "PROJECT_ID"
-    # staging_bucket = "gs://YOUR_BUCKET_NAME"
-
+    # TODO(developer): Update project, location, and staging_bucket
```
Comment (msampathkumar) [inline generative_ai/gemini_reasoning_engine.py:31]: 1. `location` in TODO is not required.
2.  Please include `# staging_bucket = "gs://YOUR_BUCKET_NAME"` example.

File: generative_ai/gemini_reasoning_engine.py [hunk 3]
```diff
@@ -153,51 +144,41 @@ def query_reasoning_engine(project_id: str, reasoning_engine_id: str) -> object:
...
+def list_reasoning_engines() -> List[reasoning_engines.ReasoningEngine]:
     # [START generativeaionvertexai_list_reasoning_engines]
     import vertexai
     from vertexai.preview import reasoning_engines
 
-    # TODO(developer): Update and un-comment below lines
-    # project_id = "PROJECT_ID"
-
-    vertexai.init(project=project_id, location="us-central1")
+    # TODO(developer): Update project_id and location
```
Comment (msampathkumar) [inline generative_ai/gemini_reasoning_engine.py:152]: `location` in TODO is not required.

File: generative_ai/gemini_reasoning_engine.py [hunk 4]
```diff
@@ -153,51 +144,41 @@ def query_reasoning_engine(project_id: str, reasoning_engine_id: str) -> object:
...
     # [START generativeaionvertexai_get_reasoning_engine]
     import vertexai
     from vertexai.preview import reasoning_engines
 
-    # TODO(developer): Update and un-comment below lines
-    # project_id = "PROJECT_ID"
-    # reasoning_engine_id = "REASONING_ENGINE_ID"
-
-    vertexai.init(project=project_id, location="us-central1")
+    # TODO(developer): Update project_id and location
```
Comment (msampathkumar) [inline generative_ai/gemini_reasoning_engine.py:166]: `location` in TODO is not required.

File: generative_ai/gemini_reasoning_engine.py [hunk 5]
```diff
@@ -153,51 +144,41 @@ def query_reasoning_engine(project_id: str, reasoning_engine_id: str) -> object:
...
     # [START generativeaionvertexai_delete_reasoning_engine]
     import vertexai
     from vertexai.preview import reasoning_engines
 
-    # TODO(developer): Update and un-comment below lines
-    # project_id = "PROJECT_ID"
-    # reasoning_engine_id = "REASONING_ENGINE_ID"
-
-    vertexai.init(project=project_id, location="us-central1")
+    # TODO(developer): Update project_id and location
```
Comment (msampathkumar) [inline generative_ai/gemini_reasoning_engine.py:180]: `location` in TODO is not required.

File: generative_ai/gemini_text_input_example.py [hunk 6]
```diff
@@ -11,17 +11,18 @@
...
+def generate_from_text_input() -> str:
     # [START generativeaionvertexai_gemini_generate_from_text_input]
     import vertexai
     from vertexai.generative_models import GenerativeModel
 
-    # TODO(developer): Update and un-comment below line
-    # project_id = "PROJECT_ID"
-
-    vertexai.init(project=project_id, location="us-central1")
+    # TODO(developer): Update project_id and location
```
Comment (msampathkumar) [inline generative_ai/gemini_text_input_example.py:24]: `location` in TODO is not required

File: generative_ai/list_tuned_models.py [hunk 7]
```diff
@@ -11,27 +11,23 @@
...
+def list_tuned_models() -> None:
     """List tuned models."""
     # [START generativeaionvertexai_sdk_list_tuned_models]
     import vertexai
 
     from vertexai.language_models import TextGenerationModel
 
-    # TODO(developer): Update values for project_id & location
-    vertexai.init(project=project_id, location=location)
+    # TODO(developer): Update project_id and location
```
Comment (msampathkumar) [inline generative_ai/list_tuned_models.py:26]: `location` in TODO is not required.

File: generative_ai/list_tuned_code_generation_models.py [hunk 8]
```diff
@@ -11,30 +11,23 @@
...
+def list_tuned_code_generation_models() -> None:
     """List tuned models."""
     # [START aiplatform_sdk_list_tuned_code_generation_models]
 
     import vertexai
     from vertexai.preview.language_models import CodeGenerationModel
 
-    # TODO(developer): Update values for project_id & location
-    vertexai.init(project=project_id, location=location)
+    # TODO(developer): Update project_id and location
```
Comment (msampathkumar) [inline generative_ai/list_tuned_code_generation_models.py:26]: `location` in TODO is not required.

**Review by msampathkumar — COMMENTED** [review]
Need minor docs updates

File: generative_ai/code_generation_function.py (hunk 1 above)
Comment (Thoughtseize1) [reply generative_ai/code_generation_function.py:21]: I suppose not. I am not just giving advice to users.
Removed it.

File: generative_ai/gemini_reasoning_engine.py (hunk 2 above)
Comment (Thoughtseize1) [reply generative_ai/gemini_reasoning_engine.py:31]: Examples added as in Set 4:
   ```
  # PROJECT_ID = "your-project-id"
  # staging_bucket = "gs://YOUR_BUCKET_NAME"
```

File: generative_ai/gemini_reasoning_engine.py [hunk 9]
```diff
@@ -73,12 +71,10 @@ def create_reasoning_engine_advanced(
     import vertexai
     from vertexai.preview import reasoning_engines
 
-    # TODO(developer): Update and un-comment below lines
-    # project_id = "PROJECT_ID"
-    # location = "us-central1"
-    # staging_bucket = "gs://YOUR_BUCKET_NAME"
-
-    vertexai.init(project=project_id, location=location, staging_bucket=staging_bucket)
+    # TODO(developer): Update project_id, location, and staging_bucket
```
[the remark this answers is not in the document]
Comment (Thoughtseize1) [reply generative_ai/gemini_reasoning_engine.py:74]: Done✅

File: generative_ai/gemini_reasoning_engine.py (hunk 3 above)
Comment (Thoughtseize1) [reply generative_ai/gemini_reasoning_engine.py:152]: Done✅

File: generative_ai/gemini_reasoning_engine.py (hunk 4 above)
Comment (Thoughtseize1) [reply generative_ai/gemini_reasoning_engine.py:166]: Done✅

File: generative_ai/gemini_reasoning_engine.py (hunk 5 above)
Comment (Thoughtseize1) [reply generative_ai/gemini_reasoning_engine.py:180]: Done✅

File: generative_ai/gemini_text_input_example.py (hunk 6 above)
Comment (Thoughtseize1) [reply generative_ai/gemini_text_input_example.py:24]: Done✅

File: generative_ai/list_tuned_models.py (hunk 7 above)
Comment (Thoughtseize1) [reply generative_ai/list_tuned_models.py:26]: Done✅

File: generative_ai/list_tuned_code_generation_models.py (hunk 8 above)
Comment (Thoughtseize1) [reply generative_ai/list_tuned_code_generation_models.py:26]: Done✅

File: generative_ai/grounding.py [hunk 10]
```diff
@@ -29,8 +32,8 @@ def grounding(
 
     from vertexai.language_models import GroundingSource, TextGenerationModel
 
-    # TODO(developer): Update values for project_id, location
-    vertexai.init(project=project_id, location=location)
+    # TODO(developer): Update project_id and location
+    vertexai.init(project=PROJECT_ID, location=LOCATION)
```
Comment (msampathkumar) [inline generative_ai/grounding.py:36]: Lets maintain consistency. Simple option is `vertexai.init(project=PROJECT_ID, location="us-central1")`

```suggestion
    vertexai.init(project=PROJECT_ID, location="us-central1")
```

File: generative_ai/grounding.py [hunk 11]
```diff
@@ -11,15 +11,18 @@
...
 # limitations under the License.
+import os
 
 from typing import Optional
 
 from vertexai.language_models import TextGenerationResponse
 
 
+PROJECT_ID = os.getenv("GOOGLE_CLOUD_PROJECT")
+LOCATION = "us-central1"
```
Comment (msampathkumar) [inline generative_ai/grounding.py:22]:
```suggestion
```

**Review by msampathkumar — COMMENTED** [review]
Almost there! 2 Minor changes are pending. Check and commit them.

File: generative_ai/grounding.py (hunk 11 above)
Comment (Thoughtseize1) [reply generative_ai/grounding.py:22]: Done✅

File: generative_ai/grounding.py (hunk 10 above)
Comment (Thoughtseize1) [reply generative_ai/grounding.py:36]: ✅ Done

**Review by msampathkumar — APPROVED** [review]
LGTM

**Pull request merged**